src/sentry/api/endpoints/organization_pinned_searches.py

```diff
@@ -44,6 +49,20 @@ def put(self, request, organization):
                 owner=request.user,
                 type=result['type'],
             )
+            try:
+                # If we pinned an existing search, return the details about that
+                # search.
+                existing_search = SavedSearch.objects.filter(
+                    Q(organization=organization, owner__isnull=True) | Q(is_global=True),
+                    type=result['type'],
+                    query=result['query'],
+                )[:1].get()
```

Comment:
Do we end up with two savedsearch rows with the same query/type and different owner values? Does the GET endpoint need to de-duplicate the pin + org search?

Reply:
Yeah, the list saved search endpoint already handles this.